test/test.js

```diff
@@ -3,77 +3,159 @@
   });
 
-  var option = { html5embed: {
-    useLinkSyntax: true,
-    templateName: "template",
-    attributes: {
-      "video": "",
-      "audio": ""
+  var option = {
+    html5embed: {
+      useLinkSyntax: true,
+      templateName: "template",
+      attributes: {
+        "video": "",
+        "audio": ""
+      }
     }
-  } };
+  };
 
   var md = require('markdown-it')().use(require('../lib'), option);
 
   generate(path.join(__dirname, 'fixtures/with-handlebars.txt'), md);
 });
 
 describe("embedding with [[html5embed]] clause", function() {
-  var options = { html5embed: {
+  var options = {
+    html5embed: {
       inline: false
-  } };
+    }
+  };
 
   var md = require('markdown-it')().use(require('../lib'), options);
 
   generate(path.join(__dirname, 'fixtures/with-placeholder-syntax.txt'), md);
 });
 
 describe("embedding with auto-append", function() {
-  var options = { html5embed: {
-    inline: false,
-    autoAppend: true
-  } };
+  var options = {
+    html5embed: {
+      inline: false,
+      autoAppend: true
+    }
+  };
 
   var md = require('markdown-it')().use(require('../lib'), options);
 
   generate(path.join(__dirname, 'fixtures/with-auto-append.txt'), md);
-});
\ No newline at end of file
+});
+
+describe('markdown-it-html5-embed with image syntax + custom messages', function() {
+  var option = {
+    html5embed: {
+      useImageSyntax: true,
+      attributes: {
+        'audio': 'width="320" controls class="audioplayer"',
+        'video': 'width="320" height="240" class="audioplayer" controls'
+      },
+      messages: customMessages
+    }
+  };
+  // Don't re-use cached function with old bindings
+  delete require.cache[require.resolve('../lib')];
+
+  var md = require('markdown-it')().use(require('../lib'), option);
+  generate(path.join(__dirname, 'fixtures/image-syntax-custom-messages.txt'), md);
+});
+
+describe('markdown-it-html5-embed with image syntax + custom translation fn', function() {
+
+  // Simply get the upper case version of the translation, if any
+  var translateFn = function(messageObj) {
+    return this[messageObj.language][messageObj.messageKey] ?
+      this[messageObj.language][messageObj.messageKey]
+      .toUpperCase()
+      .replace('%S', messageObj.messageParam) :
+      '';
+  };
+
+  var option = {
+    html5embed: {
+      useImageSyntax: true,
+      attributes: {
+        'audio': 'width="320" controls class="audioplayer"',
+        'video': 'width="320" height="240" class="audioplayer" controls'
+      },
+      messages: customMessages,
+      translateFn: translateFn
+    }
+  };
+
+  // Don't re-use cached function with old bindings
+  delete require.cache[require.resolve('../lib')];
+
+  var md = require('markdown-it')().use(require('../lib'), option);
+  var env = { language: 'de' };
+
+  // Pass along env to generated tests
+  md.orig_render = md.render;
```

Comment:
Nit: though it was my recommendation to write it this way, I would actually prefer to have it in the camel case here to comform the commonly accepted style: `origRender`